web/src/beta/components/fields/LocationField/index.tsx

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { LatLng } from "@reearth/beta/utils/value";
+import { styled } from "@reearth/services/theme";
+
+import Property from "..";
+import NumberInput from "../common/NumberInput";
+
+type Props = {
+  name?: string;
+  description?: string;
+  value?: LatLng;
+  onChange?: (location: LatLng) => void;
+};
+
+const TextInput: React.FC<Props> = ({ name, description, value, onChange }) => {
+  const [location, setLocation] = useState<LatLng>(value || { lat: 0, lng: 0 });
+
+  useEffect(() => {
+    if (value) setLocation(value);
+  }, [value]);
+
+  const handleChange = useCallback((coordination: string, newValue: number | undefined) => {
+    if (newValue === undefined) return;
+
+    setLocation(prevLocation => ({
+      ...prevLocation,
+      [coordination === "Latitude" ? "lat" : "lng"]: newValue,
+    }));
+  }, []);
+  useEffect(() => {
+    if (location) onChange?.(location);
+  }, [location, onChange]);
```

Comment:
Will this not produce an infinite loop?

Reply:
I don't think so, can you explain more why ?

Reply:
fixed